### ISSUE TITLE
Train_test_split index bug, scoring bug in SVR

### ISSUE THREAD

**Issue by nilslacroix** [intel/scikit-learn-intelex#1000]
**Describe the bug**
When I use train_test_split the indeces are shuffled sometimes, sometimes not. Also if you use the scoring method of an SVR estimator multiple times in a row with the same estimator and the same parameters, it can produce different scores.

**To Reproduce**
well you should be able to write a simple reproducer yourself, since this occurs frequently.

**Expected behavior**
No bugs.

**Environment:**
- Windows 10
-  Anaconda

**Comment by nilslacroix** [comment]
sklearnex.svm.SVR is also buggy. Using .predict() on the same X_test set in a pipeline with preprocessors and encoders yiels different predictions. Just do it like ~50 times in a row and you will see what I mean.

**Comment by jamescporter** [comment]
any update on this?

**Comment by isuruf** [comment]
I looked into this and the issue seems to be because on AMD, the code is throttled to sse2 baseline which seems to be buggy. https://github.com/uxlfoundation/oneDAL/pull/3026 should indirectly solve this as AMD hardware newer than Bulldozer will use the `sse42` code path with that.

**Comment by napetrov** [comment]
https://github.com/uxlfoundation/oneDAL/pull/3026
https://github.com/uxlfoundation/oneDAL/pull/3154

should solve this indirectly.  there are still problem with sse2 codepath but it's likely will be dropped anyway. 
First part of changes available in 2025.4 release and rest in 2025.5 which will be released in upcoming weeks

**Comment by syakov-intel** [comment]
@nilslacroix the latest release contains these changes, please check it out!
